layouts/shortcodes/past_events.html

```diff
@@ -1,17 +1,34 @@
 
-{{$currLang:=.Page.Lang }}
-{{$langBase:=index .Site.Data $currLang}}
+{{$currLang := .Page.Lang }}
+{{$langBase := index .Site.Data $currLang}}
 {{$events := (index $langBase).conferences.items}}
 
+{{ $years := slice }}
+{{ $years_with_events := dict }}
+
+{{/* Builds dictionary of years and events for that year */}}
+{{ range $events }}
+  {{ $t := time .end_date }}
+  {{ if $t.Before now }}
+    {{ $year := time.Format "2006" .start_date }}
+    {{ if not (index $years_with_events $year) }}
+      {{ $years_with_events = merge $years_with_events (dict $year slice) }}
+      {{ $years = $years | append $year }}
+    {{ end }}
+    {{ $events_for_year := index $years_with_events $year | append . }}
+    {{ $years_with_events = merge $years_with_events (dict $year $events_for_year) }}
+  {{ end }}
+{{ end }}
+
 <section class="container">
   <h2>{{ i18n "past-events-header" }}</h2>
   <ul>
-    {{ range (sort $events "date" "asc")}}
-      {{$t := time .end_date }}
-      {{ if $t.Before now }}
-        <li>
-          <a href="{{ .url }}">{{ .name }} - {{ .date }}</a>
-        </li>
+    {{ range $years }}
+      <h3>{{ . }}</h3>
+      {{ range (sort (index $years_with_events .) "start_date" "desc") }}
+          <li class="margin-left-20">
+            <a href="{{ .url }}">{{ .name }} - {{ time.Format "January 06" .start_date }}</a>
```

Comment:
This is treating it as MM-YY instead of MM-dd. Good lord is Hugo time.Format hot doo doo, having had to read it to figure out what went wrong. I'd recommend just using the `.date` in the object. That's supposed to be the localized format (if any of our community wanted to use it). Since Hugo doesn't support localized date format strings, and adding our own support would be jank as hell, I'd recommend just using that.

Reply:
Oh wow you're right, it won't localize the dates... 

Ok, I've changed it so the `.date` is used directly instead of `time.Format`.

Reply:
This gives us the option of localizing, even if it's by hand. Also, it means we don't have to use what is probably one of the worst date format implementations I've ever seen ;P